libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -410,6 +410,39 @@ bool ExpressionCompiler::visit(TupleExpression const& _tuple)
 bool ExpressionCompiler::visit(UnaryOperation const& _unaryOperation)
 {
 	CompilerContext::LocationSetter locationSetter(m_context, _unaryOperation);
+
+	if (FunctionDefinition const* function = _unaryOperation.annotation().userDefinedFunction)
+	{
+		solAssert(function->isFree(), "Only free functions can be bound to a user type operator.");
```

Comment:
Library functions should be allowed. I think we need to fix and add test case for that.

Reply:
fixed